src/contracts/middleware/IndexRegistry.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "../libraries/BN254.sol";
+import "forge-std/Test.sol";
+
+
+contract IndexRegistry is IIndexRegistry, Test {
+
+    IRegistryCoordinator public registryCoordinator;
```

Comment:
should this maybe be immutable? I don't see anywhere to change it + this contract isn't designed (for now at least) to be upgradeable.
Perhaps we should chat some about intended use and which path is better.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/8673c2394e750d95ab4d167c7beab58593768058

made it immutable. seems consistent. It needs to actually be storage in the StakeRegistry because the coordinator and stake registry are the same contract!